File: 0058-length-of-last-word/0058-length-of-last-word.java
class Solution {
    public int lengthOfLastWord(String s) {
        int end = s.length() - 1;

        while (end >= 0 && s.charAt(end) == ' ') end--;

        int wordLength = 0;
        while (end >= 0 && s.charAt(end) != ' ') {
            wordLength++;
            end--;
        }

        return wordLength;
    }
}

File: 0058-length-of-last-word/0058-length-of-last-word.kt
class Solution {
    fun lengthOfLastWord(s: String): Int {
        var end = s.length - 1
        
        while (end >= 0 && s[end] == ' ') end--

        var wordLength = 0
        while (end >= 0 && s[end] != ' ') {
            wordLength++
            end--
        }

        return wordLength
    }
}